### ISSUE TITLE
Shape Support and Understanding for multi custom loss

### ISSUE THREAD

**Issue by lschaupp** [keras-team/keras#11601]
I am using two custom loss functions within Keras but right now I am not sure if the training of the network is correct - I believe the debugging of shapes gives me the problem. So maybe I can get some feedback if my thought process behind tf shapes is correct..... I do believe my first custom loss function works well, but not so sure about my second one. So I am hoping for some input!

Network:

> Layer (type)                    Output Shape         Param #     Connected to                     
> ==================================================================================================
> input_1 (InputLayer)            (None, 48, 1030, 1)  0                                            
> __________________________________________________________________________________________________
> input_2 (InputLayer)            (None, 48, 1030, 1)  0                                            
> __________________________________________________________________________________________________
> sequential_1 (Sequential)       (None, 128)          33550208    input_1[0][0]                    
>                                                                  input_2[0][0]                    
> __________________________________________________________________________________________________
> concatenate_1 (Concatenate)     (None, 256)          0           sequential_1[1][0]               
>                                                                  sequential_1[2][0]               
> __________________________________________________________________________________________________
> dense_2 (Dense)                 (None, 1024)         263168      concatenate_1[0][0]              
> __________________________________________________________________________________________________
> activation_1 (Activation)       (None, 1024)         0           dense_2[0][0]                    
> __________________________________________________________________________________________________
> dense_3 (Dense)                 (None, 2048)         2099200     activation_1[0][0]               
> __________________________________________________________________________________________________
> activation_2 (Activation)       (None, 2048)         0           dense_3[0][0]                    
> __________________________________________________________________________________________________
> dense_yaw_output (Dense)        (None, 2)            4098        activation_2[0][0]               
> __________________________________________________________________________________________________
> lambda_1 (Lambda)               (None, 1)            0           sequential_1[1][0]               
>                                                                  sequential_1[2][0]               
> __________________________________________________________________________________________________
> output (Activation)         (None, 2)  

model = Model(inputs=[a,b], outputs = [distance, output])

My labels for the 2nd custom loss function (labels, output_labels) which go into "output" are generated like this:

```
[[ 1.         -0.41825404]
 [ 1.          0.75972638]
 [ 1.          0.05141078]
 [ 1.         -0.12097437]
 [ 1.         -1.42686069]
 [ 1.          0.37045432]
 [ 1.         -1.58479541]
 [ 1.         -0.37629725]]
```
(and returned by a generator with labels for the 1st custom loss like this:)
`return [images_left,images_right], [labels, ouput_labels]`

2nd custom loss:

```
def output_loss(y_true, y_pred):
    y_true = y_true[:] #Batch dimension access
    y_pred = y_pred[:] #Batch dimension access
    y_true_fl = y_true[:,1] # Angle Access (e.g. -0.41)
    y_true_bk = y_true[:,0] # Label Access (e.g. 1.)
    
    l2 = K.mean(K.square((K.cos(y_true_fl) - y_pred[:,0])*(y_true_bk))
                + K.square((K.sin(y_true_fl) - y_pred[:,1])*(y_true_bk)))
    return l2
```
